src/ui/components/FieldsEditor/FieldsEditor.tsx

```diff
@@ -1,35 +1,34 @@
 import { Field } from '@/model/field/Field';
-import { ReactElement, useEffect, useMemo, useState } from 'react';
+import { ReactElement, useEffect, useState } from 'react';
 import { CommandTypes, sendCommandToContent } from '@/bus/Command';
 import { SingleFieldEditor } from '@/ui/components/FieldsEditor/SingleFieldEditor';
 import { EventTypes } from '@/bus/Event';
 import { ContentEventHandler } from '@/ui/components/ContentEventHandler';
+import { Subject } from '@/model/Subject';
+import { getSchema } from '@/model/Schema';
 
 // Displays a list of fields that can be "edited" by selecting the content of each field,
 // which is done by clicking on elements in the source site.
 export function FieldsEditor( props: {
-	fields: { name: string; field: Field }[];
-	selectors: {
-		name: string;
-		selector?: string;
-	}[];
+	subject: Subject;
+	selectors?: Record< string, string >;
 	onFieldChanged: ( name: string, field: Field, selector: string ) => void;
 } ) {
-	const { fields, onFieldChanged } = props;
+	const { subject, selectors, onFieldChanged } = props;
+	const schema = getSchema( subject.type );
 	const [ fieldWaitingForSelection, setFieldWaitingForSelection ] = useState<
 		false | { field: Field; name: string }
 	>( false );
 
-	// Transform the blueprint fields into a map queryable by field name.
-	const blueprintFields = useMemo<
-		Map< string, string | undefined >
-	>( () => {
-		const map = new Map< string, string >();
-		props.selectors.forEach( ( f ) => {
-			map.set( f.name, f.selector ?? '' );
-		} );
-		return map;
-	}, [ props.selectors ] );
+	const fields = subject.fields;
+
+	// Filter out fields that are not in the schema.
+	// TODO: Move this to SubjectsApi so that a Subject is guaranteed to not have unknown fields.
```

Comment:
I haven't done this here to not conflict with #170, will do so in another PR.

Reply:
Everything looks good in the PR! This comment tripped me up though: `Filter out fields that are not in the schema.`

Why would a subject have fields that are not in the schema? Shouldn't schema already be driving that aspect as well? Or I misunderstood?

Reply:
Great question, it's not obvious when this would happen. This would only be the case when a field is removed from the schema, basically the scenario described in #159. But actually I think this has already been addressed in #170, AFAICT fields that are no longer in the schema will also not be present in the subject that is returned from `fromApiResponse()`. So I will remove this code and the TODO comment.